src/TRoomDB.cpp

```diff
@@ -670,7 +675,67 @@ void TRoomDB::auditRooms(QHash<int, int>& roomRemapping, QHash<int, int>& areaRe
         }
     }
 
-    // START OF TASK 5
+    // START OF TASK 5.0 (previously omitted) - add in areas that we have an id
+    // for but no actual TArea, at this point:
+    // * id WILL be in validUsedAreaIds
+    // * id WILL NOT be in areaRemapping, but represents a id number that cannot be
+    // remapped as a new area id
+    // * id MAY be in TRoomDB::areaNamesMap
+    // * id MAY be used by rooms which will have to be added to it at some point
+    // * WILL NOT be in TRoomDB::areas - as that is the source of the error this
+    // bit of the task being addressed here is fixing
+    if (!missingAreasNeeded.isEmpty()) {
+        if (mudlet::self()->getAuditErrorsToConsoleEnabled()) {
+            QString alertMsg = tr( "[ ALERT ] - %n area(s) detected as missing in map: adding it/them in.\n"
+                                   " Look for further messsages related to the rooms that are supposed to\n"
+                                   " be in this/these area(s)...",
+                                   "Making use of %n to allow quantity dependent message form 8-) !",
+                                   missingAreasNeeded.count() );
+            mpMap->postMessage(alertMsg);
+        }
+        mpMap->appendErrorMsgWithNoLf(tr("[ ALERT ] - %n area(s) detected as missing in map: adding it/them in.\n"
+                                         " Look for further messsages related to the rooms that is/are supposed to\n"
+                                         " be in this/these area(s)...",
+                                         "Making use of %n to allow quantity dependent message form 8-) !",
+                                         missingAreasNeeded.count() ),
+                              true);
+
+        QString infoMsg;
+        if (mudlet::self()->getAuditErrorsToConsoleEnabled()) {
+            infoMsg = tr("[ INFO ]  - The missing area(s) are now called:\n"
+                         "(ID) ==> \"name\"",
+                         "Making use of %n to allow quantity dependent message form 8-) !",
+                         missingAreasNeeded.count() );
+        }
+
+        if (missingAreasNeeded.count() > 1) {
+            // Sort the ids so that the reporting is ordered, which could be
+            // helpful if there is a large number of faults
+            std::sort(missingAreasNeeded.begin(), missingAreasNeeded.end());
+            foreach (int newAreaId, missingAreasNeeded) {
```

Comment:
Useful to know - mind you that link includes a discussion with inputs from those who are _not_ enthusiastic about the forthcoming depreciation of that element!

I note that we have, I think somewhere around 18 uses of that construct scattered around our code-base and another 11 in the latest `communi` code (Qt based IRC library for those not aware of it) which AC only plumbed in recently IIRC.

For the record, can you point me and any other interested parties to any other link you consider useful in determining which new-fangled C++ `for` loop variant to use in different circumstances, I am aware that there is more than one but for old C-coders like myself some of us may not have full cognisance of the bonuses(?) and traps(!) involved with each one... :confounded: